mitxgraders/helpers/calc/mathfuncs.py

```diff
@@ -380,10 +398,44 @@ def within_tolerance(x, y, tolerance):
     # When used within graders, tolerance has already been
     # validated as a Number or PercentageString
     if isinstance(tolerance, str):
-        # Construct percentage tolerance
-        tolerance = tolerance.strip()
-        tolerance = np.linalg.norm(x) * float(tolerance[:-1]) * 0.01
+        tolerance = np.linalg.norm(x) * percentage_as_number(tolerance)
 
     difference = x - y
 
     return np.linalg.norm(difference) <= tolerance
+
+def is_nearly_zero(x, reference, tolerance):
+    """
+    Check that x is nearly zero in comparison to reference.
+
+    Args:
+        x: number or array (np array_like)
+        reference: number or array (np array_like)
```

Comment:
Note that `reference` is only needed when using a percentage tolerance.

Reply:
Added a note.